### PR TITLE
Footprint update service, SS-417

### PR DESCRIPTION
**Pull request by dimaxano**
## Purpose
We want to be sure that the robot footprint is updated when we request that on BT side. 

## Approach
Currently, the only way to get the current footprint is to listen to `/<costmap_name>/published_footprint`. Implementing based on that was quite cumbersome so I am bringing a server for a footprint update

I'm not removing topic subscriber for footprint updates, just added a mutex to separate topic and service ways of setting a footprint
For that reason, I'm not changing rviz plugging for footprint update

### CODE REVIEW

**Comment by dimaxano** [comment]
The change isn't critical so I'll put it aside until the end of Bilka evaluation after the review are done

**Review by TanmayDeshmukh — APPROVED** [review]

**Comment by TanmayDeshmukh** [comment]
Lets keep it as draft?

**Comment by dimaxano** [comment]
Yep!
@cmrfulvius you still need to review it! :)

**Comment by fvlvivs** [comment]
> Yep! @cmrfulvius you still need to review it! :)

Shame on me..

File: nav2_costmap_2d/src/costmap_2d_ros.cpp [hunk 1]
```diff
@@ -384,6 +391,22 @@ Costmap2DROS::setRobotFootprintPolygon(
   setRobotFootprint(toPointVector(footprint));
 }
 
+void
+Costmap2DROS::updateFootprintCallback(
+  const cmr_msgs::srv::UpdateFootprint::Request::SharedPtr request,
+  cmr_msgs::srv::UpdateFootprint::Response::SharedPtr response)
+{
```
Comment (fvlvivs) [inline nav2_costmap_2d/src/costmap_2d_ros.cpp:398]: Shall `std::lock_guard<std::mutex> lock(footprint_update_mutex_)` used here too?

Comment (dimaxano) [reply nav2_costmap_2d/src/costmap_2d_ros.cpp:398]: I'd say no, as both callbacks (topic and service) call the same method inside (`setRobotFootprintPolygon`), which is protected with a mutex. Until that method is called no access to the shared recourses is made.

**Review by fvlvivs — APPROVED** [review]

**Pull request merged**